src/db.rs

```diff
@@ -0,0 +1,407 @@
+pub struct DbController {
+    pool: Pool<PostgresConnectionManager<NoTls>>,
+}
+
+#[derive(Debug, Eq, PartialEq, Clone, ToSql, FromSql)]
+#[postgres(name = "client_state")]
+pub enum ClientState {
+    #[postgres(name = "initial")]
+    Initial,
+    #[postgres(name = "pattern")]
+    Pattern,
+    #[postgres(name = "number")]
+    Number,
+    #[postgres(name = "set_city")]
+    SetCity,
+}
+
+const URL: &str = "postgres://postgres:postgres@localhost/weather_bot";
+const DELETE_CLIENT: &str = include_str!("queries/delete_client.sql");
+const GET_CITY_BY_PATTERN: &str = include_str!("queries/get_city_by_pattern.sql");
+const INSERT_CLIENT: &str = include_str!("queries/insert_client.sql");
+const IS_IN_DB: &str = include_str!("queries/is_in_db.sql");
+const MODIFY_CITY: &str = include_str!("queries/modify_city.sql");
+const MODIFY_BEFORE_STATE: &str = include_str!("queries/modify_before_state.sql");
+const MODIFY_SELECTED: &str = include_str!("queries/modify_selected.sql");
+const MODIFY_STATE: &str = include_str!("queries/modify_state.sql");
+const SEARCH_CITY: &str = include_str!("queries/search_city.sql");
+const SEARCH_CLIENT: &str = include_str!("queries/search_client.sql");
+
+impl DbController {
+    async fn pool(url: &str) -> Result<Pool<PostgresConnectionManager<NoTls>>, BotDbError> {
+        let pg_mgr = PostgresConnectionManager::new_from_stringlike(url, NoTls)?;
+
+        Ok(Pool::builder().build(pg_mgr).await?)
+    }
+
+    pub async fn new() -> Result<Self, BotDbError> {
+        let pl = Self::pool(URL).await?;
+        Ok(DbController { pool: pl })
+    }
+
+    pub async fn rollback(db_transaction: Transaction<'_>) -> Result<(), BotDbError> {
+        Ok(db_transaction.rollback().await?)
+    }
+
+    // Encrypt a String into a BYTEA
+    async fn encrypt_string(
+        some_string: String,
+        keypair: &PKey<Private>,
+    ) -> Result<Vec<u8>, BotDbError> {
+        let mut encrypter = Encrypter::new(keypair)?;
+        encrypter.set_rsa_padding(Padding::PKCS1)?;
+        let st_bytes = some_string.as_bytes();
+        let len: usize = encrypter.encrypt_len(st_bytes)?;
+        let mut encrypted = vec![0; len];
+        let encrypted_len = encrypter.encrypt(st_bytes, &mut encrypted)?;
+        encrypted.truncate(encrypted_len);
+        Ok(encrypted)
+    }
+
+    // Decrypt a BYTEA into a String
+    async fn decrypt_string(
+        encrypted: &[u8],
+        keypair: &PKey<Private>,
+    ) -> Result<String, BotDbError> {
+        let mut decrypter = Decrypter::new(keypair)?;
+        decrypter.set_rsa_padding(Padding::PKCS1)?;
+        let buffer_len = decrypter.decrypt_len(encrypted)?;
+        let mut decrypted = vec![0; buffer_len];
+        let decrypted_len = decrypter.decrypt(encrypted, &mut decrypted)?;
+        decrypted.truncate(decrypted_len);
+        Ok(String::from_utf8(decrypted)?)
+    }
+
+    pub async fn is_in_db(&self, chat_id: &i64, user_id: u64) -> Result<bool, BotDbError> {
```

Comment:
```suggestion
    pub async fn chat_exists(&self, chat_id: &i64, user_id: u64) -> Result<bool, BotDbError> {
```

Reply:
Solve in [0552d5a](https://github.com/pxp9/weather_bot_rust/pull/6/commits/0552d5a15b0c836ecd00bb26931d454e041cceaf)